core/src/main/scala/com/eharmony/spotz/optimizer/random/RandomSearch.scala

```diff
@@ -53,7 +53,12 @@ abstract class RandomSearch[P, L](
     val rng = new Random(theSeed)
     // Get rid of the first sample due to low entropy
     rng.nextDouble()
-    factory(params.map { case (label, sampler) => (label, sampler(rng)) } )
+    factory(params.foldLeft(Map.empty[String, Any]){ case (m, (label, sampler)) =>
```

Comment:
I don’t think this will work in the general case because Map doesn’t have a guaranteed iteration order since it in an Iterable and not a Seq. The consequence is that if the tests provided had more hyperparameters to search through, the test might fail because the dependent feature might be selected first. Try this with a property based test with a Map with 6 hyperparameters. A fix would be to have hyperparameters be a Seq[(String, HyperParameter)]. 

The API could accept 

(String, HyperParameter)*

and then it wouldn’t look that different.

(Sorry about formatting. On mobile)

Reply:
You are right, of course.
The API could be made to accept `(String, HP)*` but that'd be a pretty big breaking change, which is something we tried to avoid... In that case we'd probably change all the hyperparameters to always take the state param (and just ignore it in the general case).

Reply:
On the other hand, `SortedMap` can be used (it's a `Map`), but it might surprise people down the line...

Reply:
`scala.collection.immutable.SortedMap` could be a stopgap (though stacking stopgaps on top of each other isn't a good solution usually)